libs/blocks/merch/merch.js

```diff
@@ -560,6 +560,23 @@ export async function initService(force = false, attributes = {}) {
     fetchCheckoutLinkConfigs.promise = undefined;
   }
   const { commerce, env: miloEnv, locale: miloLocale } = getConfig();
+
+  const extraAttrs = [
+    'checkout-workflow-step',
+    'force-tax-exclusive',
+    'checkout-client-id',
+    'allow-override',
+  ];
+
+  extraAttrs.forEach((attr) => {
+    const camelCaseAttr = attr.replace(/-([a-z])/g, (g) => g[1].toUpperCase());
+    // eslint-disable-next-line no-prototype-builtins
+    if (commerce?.hasOwnProperty(camelCaseAttr)) {
+      const value = commerce[camelCaseAttr];
+      delete commerce[camelCaseAttr];
```

Comment:
I think we should not alter `commerce` object.

Reply:
those attributes could be added through attributes object